api-bucket-policy.go

```diff
@@ -145,3 +145,32 @@ func (c *Client) getBucketPolicy(ctx context.Context, bucketName string) (string
 	policy := string(bucketPolicyBuf)
 	return policy, err
 }
+
+// CheckObjectManagePermissions verifies if user has object permissions for given bucket
+func (c *Client) CheckObjectManagePermissions(ctx context.Context, bucket string, user string) error {
```

Comment:
And also why is this an S3 SDK API? This is specific to MinIO and must be in madmin-go

Reply:
This we are checking with replicated bucket target clients. We need to reach all targets of bucket replication and see target has got PutObject/DeleteObject etc. Is there a way to achieve it? I may be missing something here.

Reply:
this should be in madmin-go

Reply:
ok, will better move to madmin-go